sdk/trace/provider.go

```diff
@@ -350,15 +351,115 @@ func WithSampler(s Sampler) TracerProviderOption {
+//
+// • AttributeValueLengthLimit: OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT
+//
+// • AttributeCountLimit: OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT
+//
+// • EventCountLimit: OTEL_SPAN_EVENT_COUNT_LIMIT
+//
+// • AttributePerEventCountLimit: OTEL_EVENT_ATTRIBUTE_COUNT_LIMIT
+//
+// • LinkCountLimit: OTEL_SPAN_LINK_COUNT_LIMIT
+//
+// • AttributePerLinkCountLimit: OTEL_LINK_ATTRIBUTE_COUNT_LIMIT
+//
+// If both options are not provided and no environment variable is set, the
+// TracerProvider will use the default limit value:
+//
+// • AttributeValueLengthLimit: unlimited
+//
+// • AttributeCountLimit: 128
+//
+// • EventCountLimit: 128
+//
+// • AttributePerEventCountLimit: 128
+//
+// • LinkCountLimit: 128
+//
+// • AttributePerLinkCountLimit: 128
+//
+// If any filed of sl is zero or negative it will be replaced with the default
+// value for that field.
+//
+// Deprecated: Use WithRawSpanLimits instead. That option allows setting
+// unlimited and zero limits, this option does not. This option will be kept
+// until the next major version incremented release.
 func WithSpanLimits(sl SpanLimits) TracerProviderOption {
+	if sl.AttributeValueLengthLimit <= 0 {
+		sl.AttributeValueLengthLimit = DefaultAttributeValueLengthLimit
+	}
+	if sl.AttributeCountLimit <= 0 {
+		sl.AttributeCountLimit = DefaultAttributeCountLimit
+	}
+	if sl.EventCountLimit <= 0 {
+		sl.EventCountLimit = DefaultEventCountLimit
+	}
+	if sl.AttributePerEventCountLimit <= 0 {
+		sl.AttributePerEventCountLimit = DefaultAttributePerEventCountLimit
+	}
+	if sl.LinkCountLimit <= 0 {
+		sl.LinkCountLimit = DefaultLinkCountLimit
+	}
+	if sl.AttributePerLinkCountLimit <= 0 {
+		sl.AttributePerLinkCountLimit = DefaultAttributePerLinkCountLimit
+	}
 	return traceProviderOptionFunc(func(cfg tracerProviderConfig) tracerProviderConfig {
 		cfg.spanLimits = sl
 		return cfg
 	})
 }
 
+// WithRawSpanLimits returns a TracerProviderOption that will configure limits
+// as a TracerProvider's SpanLimits. These limits bound any Span created by a
+// Tracer from the TracerProvider.
+//
+// If this or WithSpanLimits are not provided, the TracerProvider will use the
+// value set for the corresponding environment variable:
+//
+// • AttributeValueLengthLimit: OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT
+//
+// • AttributeCountLimit: OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT
+//
+// • EventCountLimit: OTEL_SPAN_EVENT_COUNT_LIMIT
+//
+// • AttributePerEventCountLimit: OTEL_EVENT_ATTRIBUTE_COUNT_LIMIT
+//
+// • LinkCountLimit: OTEL_SPAN_LINK_COUNT_LIMIT
+//
+// • AttributePerLinkCountLimit: OTEL_LINK_ATTRIBUTE_COUNT_LIMIT
+//
+// If both options are not provided and no environment variable is set, the
+// TracerProvider will use the default limit value:
+//
+// • AttributeValueLengthLimit: unlimited
+//
+// • AttributeCountLimit: 128
+//
+// • EventCountLimit: 128
+//
+// • AttributePerEventCountLimit: 128
+//
+// • LinkCountLimit: 128
+//
+// • AttributePerLinkCountLimit: 128
+//
```

Comment:
```suggestion
// If this or WithSpanLimits are not provided, the TracerProvider will use
// the value set for the corresponding environment variable or its default value:
//
// • AttributeValueLengthLimit: OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT
//   Default: unlimited
//
// • AttributeCountLimit: OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT
//   Default: 128
//
// • EventCountLimit: OTEL_SPAN_EVENT_COUNT_LIMIT
//   Default: 128
//
// • AttributePerEventCountLimit: OTEL_EVENT_ATTRIBUTE_COUNT_LIMIT
//   Default: 128
//
// • LinkCountLimit: OTEL_SPAN_LINK_COUNT_LIMIT
//   Default: 128
//
// • AttributePerLinkCountLimit: OTEL_LINK_ATTRIBUTE_COUNT_LIMIT
//   Default: 128
//
// If any filed of SpanLimits is zero or negative it will be replaced with the
// default value for that field.
//
```